docs/faq.md

```diff
@@ -310,3 +310,93 @@ The currently available K/V cache quantization types are:
+5. Set [`OLLAMA_NUM_PARALLEL`](https://github.com/ollama/ollama/blob/a4f69a0191b304c204ef074ccd6523f121bfddfe/envconfig/config.go#L249) to 1.
+6. Using a smaller context buffer by reducing [`num_ctx`](https://github.com/ollama/ollama/blob/a4f69a0191b304c204ef074ccd6523f121bfddfe/docs/modelfile.md#valid-parameters-and-values:~:text=mirostat_tau%205.0-,num_ctx,-Sets%20the%20size).
+
+## How can I set the size of the context buffer?
+
+The size can be set in an API call (`"options":{"num_ctx":xxx}}`), in the ollama CLI (`/set parameter num_ctx xxx`), or configured in the model.  When using the OpenAI API compatability endpoint, setting the context size is not currently supported so it must be configured in the model.
+
+There are two ways to configure the context size in a model.  First, use the `/save` command in the ollama CLI:
+```console
+ollama run model-with-normal-context
+>>> /set parameter num_ctx 4096
+>>> /save model-with-4096-context
+>>> /bye
+```
+Second, create a new Modelfile:
+```console
+ollama show --modelfile model-with-normal-context > Modelfile
+echo PARAMETER num_ctx 4096 > Modelfile
+ollama create model-with-4096-context
+```
+After the model has been created, use it in the normal way:
+```console
+curl localhost:11434/api/generate -d "{\"model\":\"model-with-4096-context\",\"prompt\":\"why is the sky blue?\"}"
+```
+```console
+ollama run model-with-4096-context
+```
+If the model is large, the second approach above can cause a lot of disk activity as ollama loads the GGUF file.  This can be prevented by editing the Modelfile, uncommenting the first `FROM` statement and commenting the second `FROM` statement.
+
+## Why does the runner use 4 times the context buffer I configured?
+
+Ollama can do parallel completions, controlled by `OLLAMA_NUM_PARALLEL`.  If unset, the default is 4 (or 1 if your system has only a small amount of free RAM).  Each completion instance has its own context buffer, size given by `num_ctx`.  The value of `ctx-size` on the runner command line or the value `n_ctx` in the logs is the total context buffer that is allocated, `OLLAMA_NUM_PARALLEL` * `num_ctx`.
+
+## What is K-shift and why did it kill ollama?
+
+When the context buffer fills up during completion, the inference engine wants to shift the buffer to make room for new tokens.  Some model architectures, notably Deepseek, do not support this feature and the runner will exit with an error.  To prevent this, the context buffer (`num_ctx`) needs to be sized such that it can contain both input tokens and output tokens.  To prevent the completion from generating more output tokens than expected, `num_predict` should be set.  For example, if the model is expected to received up to 10,000 tokens and generate no more than 5,000 tokens in response, `num_ctx` is set to 15,000 and `num_predict` to 5,000.  These parameters can be set in the `options` field of an API call, or configured directly in the model.
```

Comment:
I'd call this k-shift feature a bug, I'll poke Jesse to take a look at the issue. I'd rather fix it than document it if possible.